lib/session.go

```diff
@@ -31,11 +31,35 @@ func (e *Session) Assume(arn string) (*Session, error) {
 	}
 
 	duration := expiration.Sub(time.Now())
-	creds, err := e.AWSCreds.AssumeRole(arn, duration)
+
+	var creds *AWSCredentials
+	var err error
+	var arnStruct *ARN
+
+	arnStruct, err = ParseARN(arn)
 	if err != nil {
 		return nil, err
 	}
 
+	if arnStruct.ValidRole() {
```

Comment:
This condition is incorrect. As written, this will take a valid role ARN, such as `arn:aws:iam::111222333444:role/SuperUser` and rewrite it to be `arn:aws:iam::111222333444:role/arn:aws:iam::111222333444:role/SuperUser` (an amalgamation of the caller identity and the full role ARN) which then fails when calling `AssumeRole`.

Additionally, the `if err != nil` condition above prevents bare role names from being passed far enough to attempt to calculate the role's ARN, returning a `Not an ARN` error.